src/code_generators.cpp

```diff
@@ -85,19 +85,51 @@ const char *BaseGenerator::FlatBuffersGeneratedWarning() {
 std::string BaseGenerator::NamespaceDir(const Parser &parser,
                                         const std::string &path,
                                         const Namespace &ns) {
+  return NamespaceDir(parser, path, ns, false);
+}
+
+std::string BaseGenerator::NamespaceDir(const Parser &parser,
+                                        const std::string &path,
+                                        const Namespace &ns,
+                                        const bool dasherize) {
   EnsureDirExists(path);
   if (parser.opts.one_file) return path;
   std::string namespace_dir = path;  // Either empty or ends in separator.
   auto &namespaces = ns.components;
   for (auto it = namespaces.begin(); it != namespaces.end(); ++it) {
-    namespace_dir += *it + kPathSeparator;
+    auto component = *it;
+    if (dasherize)
+      component = ToDasherizedCase(component);
+    namespace_dir += component + kPathSeparator;
```

Comment:
It could be rewritten using the ternary operator, for example:
```cpp
namespace_dir +=  !dasherize ? *it : ToDasherizedCase(*it);
namespace_dir +=  kPathSeparator;
```